receiver/otlpreceiver/config.go

```diff
@@ -21,7 +21,7 @@ const (
 	protoHTTP = "protocols::http"
 )
 
-type httpServerSettings struct {
+type HTTPServerSettings struct {
```

Comment:
I think we should call it `HTTPConfig` to avoid having `Protocols.HTTPServerSettings.HTTPServerSettings`. Also, we have https://github.com/open-telemetry/opentelemetry-collector/issues/6767, which would require this one to change later if we keep it like this

cc @mx-psi

Reply:
Makes sense to tackle this now before it becomes a breaking change, specially in light of #6767. `HTTPConfig` makes sense to me as a new name

Reply:
Done.